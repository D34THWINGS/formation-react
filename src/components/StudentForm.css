.studentForm {
  background: #333;
  margin-bottom: 2rem;
  padding: 0.5rem;
  border-radius: 5px;
}

.studentForm__title {
  color: #fff;
  margin: 0 0 1rem 0;
}

.studentForm input {
  display: inline-block;
  border: none;
  background: #fff;
  margin-right: 0.5rem;
  padding: 0.5rem;
  font-size: 1.3rem;
}

.studentForm__button {
  background: #fff;
  padding: 0.5rem;
  border: none;
  font-size: 1.3rem;
  font-weight: bold;
  cursor: pointer;
}
