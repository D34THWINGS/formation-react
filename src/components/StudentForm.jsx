import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const StyledStudentForm = styled.form`
  background: #333;
  margin-bottom: 2rem;
  padding: 0.5rem;
  border-radius: 5px;
`;

const StudentFormTitle = styled.h2`
  color: #fff;
  margin: 0 0 1rem 0;
`;

const StudentFormInput = styled.input`
  display: inline-block;
  border: none;
  background: #fff;
  margin-right: 0.5rem;
  padding: 0.5rem;
  font-size: 1.3rem;
`;

const StudentFormButton = styled.button`
  background: #fff;
  padding: 0.5rem;
  border: none;
  font-size: 1.3rem;
  font-weight: bold;
  cursor: pointer;
`;

class StudentForm extends Component {
  state = {
    firstName: '',
    lastName: '',
  };

  handleChange = input => e => this.setState({ [input]: e.target.value });

  handleSubmit = (e) => {
    e.preventDefault();
    const { firstName, lastName } = this.state;
    const { onAddStudent } = this.props;
    onAddStudent({ firstName, lastName });
    this.setState({ firstName: '', lastName: '' });
  };

  render() {
    const { firstName, lastName } = this.state;
    return (
      <StyledStudentForm onSubmit={this.handleSubmit}>
        <StudentFormTitle>Nouvel Ã©tudiant</StudentFormTitle>
        <StudentFormInput type="text" name="firstName" value={firstName} onChange={this.handleChange('firstName')} />
        <StudentFormInput type="text" name="lastName" value={lastName} onChange={this.handleChange('lastName')} />
        <StudentFormButton type="submit">Ajouter</StudentFormButton>
      </StyledStudentForm>
    );
  }
}

StudentForm.propTypes = {
  onAddStudent: PropTypes.func.isRequired,
};

export default StudentForm;
