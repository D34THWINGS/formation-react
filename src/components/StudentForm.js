import React, { Component } from 'react';

import './StudentForm.css';

class StudentForm extends Component {
  state = {
    firstName: '',
    lastName: '',
  };

  handleChange = input => e => this.setState({ [input]: e.target.value });

  handleSubmit = (e) => {
    e.preventDefault();
    const { firstName, lastName } = this.state;
    this.props.onAddStudent({ firstName, lastName });
    this.setState({ firstName: '', lastName: '' });
  };

  render() {
    const { firstName, lastName } = this.state;
    return (
      <form className="studentForm" onSubmit={this.handleSubmit}>
        <h2 className="studentForm__title">Nouvel Ã©tudiant</h2>
        <input type="text" name="firstName" value={firstName} onChange={this.handleChange('firstName')} />
        <input type="text" name="lastName" value={lastName} onChange={this.handleChange('lastName')} />
        <button type="submit" className="studentForm__button">Ajouter</button>
      </form>
    );
  }
}

export default StudentForm;
