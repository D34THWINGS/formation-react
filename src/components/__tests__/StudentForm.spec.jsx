import React from 'react';
import { shallow } from 'enzyme';

import StudentForm from '../StudentForm';

describe('StudentForm', () => {
  it('should render student form', () => {
    const wrapper = shallow(<StudentForm onAddStudent={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle input change', () => {
    const wrapper = shallow(<StudentForm onAddStudent={jest.fn()} />);
    wrapper.find('input').first().simulate('change', { target: { value: 'azerty' } });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle form submit', () => {
    const submitFn = jest.fn();
    const wrapper = shallow(<StudentForm onAddStudent={submitFn} />);
    wrapper.setState({ firstName: 'foo', lastName: 'bar' });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(submitFn).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
