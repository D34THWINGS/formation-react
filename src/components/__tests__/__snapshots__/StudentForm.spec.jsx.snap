// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentForm should handle form submit 1`] = `
<form
  className="studentForm"
  onSubmit={[Function]}
>
  <h2
    className="studentForm__title"
  >
    Nouvel étudiant
  </h2>
  <input
    name="firstName"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="lastName"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="studentForm__button"
    type="submit"
  >
    Ajouter
  </button>
</form>
`;

exports[`StudentForm should handle input change 1`] = `
<form
  className="studentForm"
  onSubmit={[Function]}
>
  <h2
    className="studentForm__title"
  >
    Nouvel étudiant
  </h2>
  <input
    name="firstName"
    onChange={[Function]}
    type="text"
    value="azerty"
  />
  <input
    name="lastName"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="studentForm__button"
    type="submit"
  >
    Ajouter
  </button>
</form>
`;

exports[`StudentForm should render student form 1`] = `
<form
  className="studentForm"
  onSubmit={[Function]}
>
  <h2
    className="studentForm__title"
  >
    Nouvel étudiant
  </h2>
  <input
    name="firstName"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="lastName"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="studentForm__button"
    type="submit"
  >
    Ajouter
  </button>
</form>
`;
